lib/http.c

```diff
@@ -1922,7 +1927,11 @@ CURLcode Curl_add_timecondition(const struct connectdata *conn,
             tm->tm_min,
             tm->tm_sec);
 
+#ifndef USE_HYPER
   result = Curl_dyn_add(req, datestr);
+#else
+  result = Curl_hyper_header(data, req, datestr);
```

Comment:
It looks like in `c-hyper` L819, the `hyper_request *` is passed to `Curl_add_timecondiction`, which is forwarded directly to `Curl_hyper_header` which expects a `hyper_headers *`.

I'm not as used to C, but could it be the `void *` coercion of the argument makes it so the compiler doesn't notice?

Reply:
Ah, you're right. Thanks! I should make these ones not use `void *` as then I would've caught my own mistake...

Reply:
One of the downsides with `hyper.h` using typedef'ed structs I think, as I can't forward-declare them. I would have to include `hyper.h` in http.c as well, which I've tried to avoid...